src/Functions/initialQueryStartTime.cpp

```diff
@@ -0,0 +1,47 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Interpreters/Context.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <Core/Field.h>
+
+namespace DB
+{
+class FunctionInitialQueryStartTime : public IFunction
+{
+    const time_t initial_query_start_time;
+public:
+    static constexpr auto name = "initialQueryStartTime";
+    static FunctionPtr create(ContextPtr context)
+    {
+        return std::make_shared<FunctionInitialQueryStartTime>(context->getClientInfo().initial_query_start_time);
+    }
+
+    explicit FunctionInitialQueryStartTime(const time_t & initial_query_start_time_) : initial_query_start_time(initial_query_start_time_) {}
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeDateTime>();
+    }
+
+    bool isDeterministic() const override { return false; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName &, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        return DataTypeDateTime().createColumnConst(
+                input_rows_count,
+                static_cast<UInt64>(initial_query_start_time));
+    }
+};
+
+REGISTER_FUNCTION(InitialQueryStartTime)
```

Comment:
Perhaps, you could add embedded documentation for the `system.functions`? The example is in https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/sin.cpp#L14-L24

```
SELECT *
FROM system.functions
WHERE name = 'sin'

Query id: e01ce267-0def-4bdc-8a89-da8b96b7d1b3

Row 1:
──────
name:             sin
is_aggregate:     0
case_insensitive: 1
alias_to:
create_query:
origin:           System
description:      Returns the sine of the argument.
syntax:           sin(x)
arguments:        - x:The number whose sine will be returned. (U)Int*, Float* or Decimal*.

returned_value:   The sine of x.
examples:         simple:

``` sql
SELECT sin(1.23)
```

``` text
0.9424888019316975
```

categories:       Mathematical, Trigonometric
```